packages/editor-web-component/src/editor-web-component.tsx

```diff
@@ -0,0 +1,89 @@
+import { SerloEditor } from '@serlo/editor'
+import React from 'react'
+import * as ReactDOM from 'react-dom/client'
+
+import {
+  exampleInitialState,
+  isInitialState,
+  type InitialState,
+} from './initial-state'
+
+// Could probably remove the export entirely, as the customElement is registered
+// below.
+export class EditorWebComponent extends HTMLElement {
+  private reactRoot: ReactDOM.Root | null = null
```

Comment:
for a lack of a better place to put this, I'll have it at the react dom root:

In vanilla JS applications, there is no `process.env.`. However, the bundle assumes that it exists and bears `"production"` or `"staging"`. Easiest workaround is to include `process = {env: 'production'}` in the banner of the bundle if the environment does not already worry about the global process variable through another bundler. The "real" solution would be not to rely on that variable in the bundle's output but I'm not sure as to how realistic that is on the short term. 

From experience, that's usually a big pain point for package distributors since bundling issues hold back adoption. Putting a workaround in one's own code looks super ugly and is frowned upon, still holding back adoption. On the other hand, even excalidraw has this issue since they also bundle react in their component, but notably they don't pretend to be agnostic and just deliver a react component.

Reply:
sorry missed github notifications... not too familiar with vite's build system but if you can add global constants that way, that would probably be good. Anywhere that's not in the actual code because it's only for "glueing stuff together"

Reply:
sorry completely missed github notifications. I'm not too familiar with vite's build system. That being said, anywhere that's at compile time to inject the thing into the final bundle. Build config sounds fair enough to me without knowing details about vite stuff.